docs/3-transfer.md

```diff
@@ -0,0 +1,271 @@
+# CEP-18 Token Transfers and Allowances
+
+This document describes how to transfer CEP-18 tokens on a Casper network using the Casper client. The [Exploring the CEP18 Contracts](./query.md) documentation contains more in depth explanations on how to find the various hashes and URefs referenced throughout this document.
+
+## Transferring CEP-18 Tokens to Another Account
+
+The following command will invoke the `transfer` entry point on your instance of CEP-18, directing it to transfer 10 of the associated CEP-18 tokens to another account.
+
+```bash
+casper-client put-deploy -n http://<node IP>:<PORT> \
+// The chain name of the Casper network on which your CEP-18 instance was installed.
+--chain-name <CHAIN NAME>\
+// The local path to your account's secret key.
+--secret-key ~/casper/demo/user_a/secret_key.pem \
+// The contract hash of your CEP-18 contract instance.
+--session-hash hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180 \
+// The name of the entry point you are invoking.
+--session-entry-point "transfer" \
+// The account hash of the account that you are sending CEP-18 tokens to.
+--session-arg "recipient:key='account-hash-9f81014b9c7406c531ebf0477132283f4eb59143d7903a2fae54358b26cea44b" \
+// The amount of CEP-18 tokens you are sending to the receiving account.
+--session-arg "amount:u256='10'" \
+// The gas payment you are allotting, in motes.
+--payment-amount "10000000000"
+```
+
+<details>
+<summary><b>Casper client command without comments</b></summary>
+
+```bash
+casper-client put-deploy -n http://<node IP>:<PORT> \
+--chain-name <CHAIN NAME>\
+--secret-key ~/casper/demo/user_a/secret_key.pem \
+--session-hash hash-b568f50a64acc8bbe43462ffe243849a88111060b228dacb8f08d42e26985180 \
+--session-entry-point "transfer" \
+--session-arg "recipient:key='account-hash-9f81014b9c7406c531ebf0477132283f4eb59143d7903a2fae54358b26cea44b" \
+--session-arg "amount:u256='50'" \
+--payment-amount "10000000000"
+```
+
+</details>
+
+This command will return a deploy hash that you can query using `casper-client get-deploy`. Querying the Deploy allows you to verify execution success, but you will need to use the `check_balance_of` entry point on the utility contract to verify the account's balance.
```

Comment:
I am not sure if this is helpful since to check balance of the user, the checker needs to pay gas fee for it, I bet on nobody won't do such thing :),

 the purpose of `cep18_test_contract` was to check variables in Rust EE level and currently not sure are used in integration test after whole refactoring.

cc @deuszex @hoffmannjan

Reply:
The purpose of the `cep18_test_contract` is to test the functioning of getter functions that would otherwise not be possible from cli since the getters only return values into contracts. They are still used.